test/dynamo/test_modules.py

```diff
@@ -2733,7 +2733,8 @@ def fn(x, mod):
         fn(inp, mod)
         self.assertEqual(num_compiles, 2)
 
-    def test_no_guard_on_torch_nn_modules(self):
+    @patch.object(torch._dynamo.config, "guard_nn_modules", True)
```

Comment:
The discrepancy was introduced by:
https://github.com/pytorch/pytorch/blob/605dfd8fb4672beef93b385ff91df5f349f2d394/torch/_dynamo/config.py#L101